### DISCUSSION TITLE
Rookie Question: Initial and Boundary Conditions

### DISCUSSION THREAD

**Discussion by cmrozoff** [ufs-community/ufs-srweather-app#531] [Q&A] [answered]
Hi all,
I'm embarrassed to ask such a fundamental question, but I hope it's a rather easy one to answer and will help others.

I'm an "extreme" rookie on running the SRW App. I'm trying to run the latest version that I've compiled on Cheyenne. The workflow successfully does things like creating a grid, surface climatology, and other odds and ends but I'm getting hung up on initial and boundary conditions. The best I can tell, my problem appears mostly that there are incomplete GFS datasets for the 6/15/19 test case to set up initial/boundary conditions on Cheyenne (for those working on the system, the files that are supposed to be located at /glade/p/ral/jntp/.... Because I simply want to run a test case in my first baby steps of running a new model (plenty of experience with WRF, CM1, etc.), the 2019 case would be fine with me, but is there a place to locate those data or other test cases to work with?  Eventually, I'll need to learn this anyway, but any training wheels like a successful test case will certainly help me get there faster.
Thanks,
Chris

**Comment by gspetro-NOAA** [comment]
Hi Chris,

Just a few questions to help us get you an answer as quickly as possible: 
- Are you working on the `develop` branch or a `release` branch of the app? (If release, which one? v2.1.0?)
- Are you using the GNU or Intel compiler?
- Is the forecast failing on the `get_extrn_ics`/`get_extrn_lbcs` task or the `make_ics`/`make_lbcs` task? 
- Could you post a screenshot of the error message from the log file (found in the `log` subdirectory of your experiment directory)? It may also be helpful to see a screenshot of your `config.yaml` file to make sure the data sections are configured correctly. 

In the meantime, you can try downloading the data locally from https://noaa-ufs-srw-pds.s3.amazonaws.com/current_srw_release_data/gst_data.tgz and pointing to that copy of the ICS/LBCS in your config file ([Chapter 7 of the User's Guide](https://ufs-srweather-app.readthedocs.io/en/develop/InputOutputFiles.html#downloading-and-staging-input-data) has some more info on getting data). If the experiment still doesn't find the data, then the problem is likely elsewhere (e.g., the config file may not be configured correctly). The log files and config file can help us figure that out!

Also, FWIW, I was a complete SRW App rookie this time last year! There's a learning curve, but once you get it to run a few times, the troubleshooting process you took to get through those first few runs will help you to run/troubleshoot successfully on your own in the future! :)

Best,
Gillian

**Comment by cmrozoff** [comment]
Dear Gillian,
Thank you so much for taking the time to respond and the encouragement. I think your link to the Cloud data for GFS ICs/LBCs may prove most propitious to a positive outcome, so I will try to obtain the data there to see what happens and report back.

In the meanwhile, I can answer your questions:
1. I am running https://github.com/ufs-community/ufs-srweather-app version (which is labeled SRWA Version 1.0.0-development in my build directory)
2. The SRW is compiled using Intel compilers
3. The forecast  is failing on the get_extrn_ics/get_extrn_lbcs task
4. The attached screenshots are (a) error message in the get_extrn_LBCs log message and the config.yaml file.

Thanks again!
Chris
<img width="868" alt="Screen Shot 2023-01-09 at 10 07 19 AM" src="https://user-images.githubusercontent.com/20117236/211366734-1ef5c63b-ba0c-4c26-a708-64dc6612ede6.png">
<img width="644" alt="Screen Shot 2023-01-09 at 10 07 51 AM" src="https://user-images.githubusercontent.com/20117236/211366736-227b1964-2359-4174-989f-7543deb95230.png">

**Reply by gspetro-NOAA** [reply, marked as answer]
Hi Chris,

Np! It looks like you may have an old/alternative data location which is causing the problem. The updated locations for the develop branch data can be found in [Ch. 5 of the User's Guide](https://ufs-srweather-app.readthedocs.io/en/develop/RunSRW.html#download-and-stage-the-data). For Cheyenne, the ICS/LBCS data would be here: /glade/p/ral/jntp/UFS_SRW_App/develop/input_model_data/ . You can make sure to set the data location by adding the following lines to the `task_get_extrn_ics:` section (in `config.yaml`): 

```
USE_USER_STAGED_EXTRN_FILES: true
EXTRN_MDL_SOURCE_BASEDIR_ICS: /glade/p/ral/jntp/UFS_SRW_App/develop/input_model_data/FV3GFS/grib2/${yyyymmddhh}
```
And these lines to the `task_get_extrn_lbcs:` section:

```
USE_USER_STAGED_EXTRN_FILES: true
EXTRN_MDL_SOURCE_BASEDIR_ICS: /glade/p/ral/jntp/UFS_SRW_App/develop/input_model_data/FV3GFS/grib2/${yyyymmddhh}
```
Another pointer: 
`EXPT_SUBDIR` is usually just the name you want the subdirectory to have rather than the whole file path. The setting you currently have in your config file may cause problems later because the workflow sets `EXPTDIR="${EXPT_BASEDIR}/${EXPT_SUBDIR}"`. If you don't want `EXPT_BASEDIR` to have the default value, you can set it in `config.yaml`. To learn a little more about these two parameters, take a look at the [Configuration Parameters Chapter](https://ufs-srweather-app.readthedocs.io/en/develop/ConfigWorkflow.html#directory-parameters) of the User's Guide. 

Hopefully these pointers will help you resolve the issue and get that forecast running, but if you continue to have issues, don't hesitate to reach out with a screenshot of the new log files!

Best,
Gillian

**Reply by cmrozoff** [reply]
This was outstanding! The model is now numerically integrating forward in time on Cheyenne (with appropriate boundary/initial conditions having been successfully handled/ingested). I appreciate your help!
